integrations/fastembed/tests/test_fastembed_ranker.py

```diff
@@ -0,0 +1,235 @@
+        assert ranker.batch_size == 50
+        assert ranker.parallel == 1
+        assert ranker.local_files_only
+        assert ranker.meta_fields_to_embed == ["test_field"]
+        assert ranker.meta_data_separator == " | "
+
+    def test_embed_incorrect_input_format(self):
+        """
+        Test for checking incorrect input format when creating embedding.
+        """
+        ranker = FastembedRanker(model_name="Xenova/ms-marco-MiniLM-L-12-v2")
+        ranker.warm_up()
+
+        query = "query"
+        string_input = "text"
+        list_integers_input = [1, 2, 3]
+
+        with pytest.raises(
+            TypeError,
+            match="FastembedRanker expects a list of Documents as input.",
+        ):
+            ranker.run(query=query, documents=string_input)
+
+        with pytest.raises(
+            TypeError,
+            match="FastembedRanker expects a list of Documents as input.",
+        ):
+            ranker.run(query=query, documents=list_integers_input)
+
+    def test_embed_metadata(self):
+        """
+        Test for checking output dimensions and embedding dimensions for documents
+        with a custom instruction and metadata.
+        """
+        ranker = FastembedRanker(
+            model_name="model_name",
+            meta_fields_to_embed=["meta_field"],
+        )
+        ranker.ranker = MagicMock()
+
+        documents = [Document(content=f"document-number {i}", meta={"meta_field": f"meta_value {i}"}) for i in range(5)]
+        query = "test"
+        ranker.run(query=query, documents=documents)
+
+        ranker.ranker.rerank.assert_called_once_with(
+            query=query,
+            documents=[
+                "meta_value 0\ndocument-number 0",
+                "meta_value 1\ndocument-number 1",
+                "meta_value 2\ndocument-number 2",
+                "meta_value 3\ndocument-number 3",
+                "meta_value 4\ndocument-number 4",
+            ],
+            batch_size=64,
+            parallel=None,
+        )
+
+    @pytest.mark.integration
+    def test_run(self):
+        ranker = FastembedRanker(model_name="Xenova/ms-marco-MiniLM-L-6-v2", top_k=2)
+        ranker.warm_up()
+
+        query = "Who is maintaining Qdrant?"
+        documents = [
+            Document(
+                content="This is built to be faster and lighter than other embedding \
+libraries e.g. Transformers, Sentence-Transformers, etc."
+            ),
+            Document(content="This is some random input"),
+            Document(content="fastembed is supported by and maintained by Qdrant."),
+        ]
+
+        result = ranker.run(query=query, documents=documents)
+
+        assert len(result["documents"]) == 2
+        first_document = result["documents"][0]
+
+        assert isinstance(first_document, Document)
+        assert first_document.content == "fastembed is supported by and maintained by Qdrant."
+        assert first_document.score == 5.472434997558594
```

Comment:
Let's only check that first document score > second document score.
Otherwise, this test can become flaky due to temporary circumstances and external changes.

Reply:
Done